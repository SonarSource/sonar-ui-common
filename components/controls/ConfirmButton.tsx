/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import ConfirmModal, { ConfirmModalProps } from './ConfirmModal';
import ModalButton, { ChildrenProps, ModalProps } from './ModalButton';

interface Props<T> extends ConfirmModalProps<T> {
  children: (props: ChildrenProps) => React.ReactNode;
  modalBody: React.ReactNode;
  modalHeader: string;
  modalHeaderDescription?: React.ReactNode;
}

interface State {
  modal: boolean;
}

export default class ConfirmButton<T> extends React.PureComponent<Props<T>, State> {
  renderConfirmModal = ({ onClose }: ModalProps) => {
    const {
      children,
      modalBody,
      modalHeader,
      modalHeaderDescription,
      ...confirmModalProps
    } = this.props;
    return (
      <ConfirmModal
        header={modalHeader}
        headerDescription={modalHeaderDescription}
        onClose={onClose}
        {...confirmModalProps}>
        {modalBody}
      </ConfirmModal>
    );
  };

  render() {
    return <ModalButton modal={this.renderConfirmModal}>{this.props.children}</ModalButton>;
  }
}
