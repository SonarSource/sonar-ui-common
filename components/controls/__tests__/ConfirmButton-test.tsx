/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import ConfirmButton from '../ConfirmButton';

it('should display a modal button', () => {
  expect(shallowRender()).toMatchSnapshot();
});

it('should display a confirm modal', () => {
  expect(
    shallowRender().find('ModalButton').prop<Function>('modal')({ onClose: jest.fn() })
  ).toMatchSnapshot();
});

function shallowRender() {
  return shallow(
    <ConfirmButton
      confirmButtonText="submit"
      modalBody={<div />}
      modalHeader="title"
      onConfirm={jest.fn()}>
      {() => 'Confirm button'}
    </ConfirmButton>
  );
}
