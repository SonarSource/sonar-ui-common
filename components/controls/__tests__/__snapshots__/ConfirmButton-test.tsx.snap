// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a confirm modal 1`] = `
<ConfirmModal
  confirmButtonText="submit"
  header="title"
  onClose={[MockFunction]}
  onConfirm={[MockFunction]}
>
  <div />
</ConfirmModal>
`;

exports[`should display a modal button 1`] = `
<ModalButton
  modal={[Function]}
>
  <Component />
</ModalButton>
`;
